lib/cosmos/interfaces/protocols/terminated_stream_protocol.rb

```diff
@@ -65,8 +65,12 @@ def reduce_to_single_packet
           else
             packet_data = @data[0..(index + @read_termination_characters.length - 1)]
           end
-        else
-          packet_data = ''
+        else # @data begins with the termination characters
+          if @strip_read_termination
+            packet_data = ''
+          else # Keep everything
+            packet_data = @data[0..-1]
```

Comment:
The termination characters IS all the data. That's what the comment above says because index is 0